sky/skylet/job_lib.py

```diff
@@ -342,8 +365,10 @@ def update_job_status(job_owner: str,
                 # that value. So we take the max here to keep it at RUNNING.
                 status = max(status, original_status)
                 _set_status_no_lock(job_id, status)
```

Comment:
To ensure we only ever update to RUNNING once, how about we change these to
```
if status != original_status:  # Prevents redundant update.
    _set_status_no_lock(job_id, max(status, original_status))
```

Reply:
That is a good point! I added the assertion back the `set_status` function. Thanks!